lib/htmllint.js

```diff
@@ -23,6 +23,21 @@ module.exports = function( config, done ) {
     return str;
   }
 
+  // parse and, if needed, normalize error messages from HttpClient to vnu.jar format
+  function parseErrorMessages( errors ) {
+    var parsed = JSON.parse( config.server ? '[' + errors.trim().replace( /\n/g, ',' ) + ']' : errors );
+    var messages = parsed.messages;
+    if ( config.server ) {
+      messages = [].concat.apply([], parsed.map(function( file ) {
```

Comment:
The JSON returned by the servlet looks slightly different than normal java -jar execution.
java -jar:
```
{
    "messages": [ // all messages for all files
        { "message": ..., "type": ..., "url": ..., ... },
        ...
    ]
}
```
servlet:
```
{
    "language": "en",
    "messages": [ // all messages for a single file
        { "message": ..., "type": ..., ... }, // no "url" property
        ...
    ],
    "url": ...
} // one json object per file, concatenated with a newline
{
    ...
}
```
This necessitates the `'[' + errors.trim().replace( /\n/g, ',' ) + ']'`.
`parsed.map` extracts the "message" property, resulting in an array of arrays, instead of an array of objects, which is then flattened by the concatenation.
One might specify `Array.prototype.concat.apply` instead of `[].concat`, which is more verbose, but perhaps more clear.